### PR TITLE
adjust sequence of eventf for creation of resources

### PR DESCRIPTION
**Pull request by jichenjc**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by jichenjc** [comment]
recheck

File: pkg/cloud/services/networking/network.go [hunk 1]
```diff
@@ -139,6 +138,7 @@ func (s *Service) ReconcileNetwork(clusterName string, openStackCluster *infrav1
 		}
 	}
 
+	record.Eventf(openStackCluster, "SuccessfulCreateNetwork", "Created network %s with id %s", networkName, network.ID)
```
Comment (hidekazuna) [inline pkg/cloud/services/networking/network.go:141]: If adding tags fails, network is created but no event recorded.
Current event recording is better.

Comment (sbueringer) [reply pkg/cloud/services/networking/network.go:141]: I agree. If it's really important that we have another event after the tagging we could consider producing two events. One after network creation and one after tagging. But I'm not sure what the use case is for this and thus what we can achieve with it.

**Comment by sbueringer** [comment]
@jichenjc do we need this PR *and* https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/828?

**Comment by jichenjc** [comment]
I think we can use that PR now

**Pull request closed without merging**